### ISSUE TITLE
Keep unlocked SSH keys in cache for duration of the session

### ISSUE THREAD

**Issue by jntesteves** [Murmele/Gittyup#497]
My SSH key has a password. Currently, the app requires me to type the password again and again for each remote of each repository. I have repositories with some 6 remotes, which means typing the password 6 times just to fetch everything. Since all remotes and all repositories use the exact same SSH key, the unlocked key could be cached in memory until the app is closed, and reused for all remotes and all repositories that have the same key. That's what SmartGit does.

**Comment by jntesteves** [comment]
I'm on Linux v1.2.2, the latest on Flathub. I don't have the option in your screenshot.

![image](https://user-images.githubusercontent.com/133886/226973581-3fafad0a-a66e-414a-9ed2-6607ad4389c6.png)

**Comment by Murmele** [comment]
Can you try the latest release?
https://github.com/Murmele/Gittyup/releases/tag/development

**Comment by jntesteves** [comment]
Hi, I'm sorry for the long delay in responding, I didn't want to add noise to this issue until I figure some things up. I tested the development version and still had the same problem. Also, the git-credential-* plugins that come installed by default in most distros and in the Flatpak runtime for this app are useless, they are no more than examples and shouldn't be used in production.

But none of that matters, my problem was something else entirely. Somehow I've got wrong permissions (o711) on my SSH key. I found out when trying to use git on the command line and the SSH agent just plain refused to work, saying the key with these permissions was insecure. I have no idea how that happened, but I could easily fix that with `chmod a-x` my SSH key.

Now, with my SSH key fixed, Gittyup is reading my SSH key from the OS's keychain, where the password is saved encrypted. I rolled back the app to the stable version and this is still true. I don't even need to use the **Credentials** option, the git-credential-* plugins are seemingly unnecessary on Linux. Given this is already the best possible solution, I'm closing this issue.
